packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -34,64 +36,140 @@ export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
     density,
     ...otherProps
   } = props;
+
   let ref = useRef<HTMLDivElement>();
+  let wrapperRef = useRef<HTMLDivElement>();
   let state = useSingleSelectListState<T>({
     ...props,
     onSelectionChange
   });
 
+  let {direction} = useLocale();
   let {styleProps} = useStyleProps(otherProps);
   let {tabListProps, tabPanelProps} = useTabs(props, state, ref);
+  let [collapse, setCollapse] = useValueEffect(false);
   let [selectedTab, setSelectedTab] = useState<HTMLElement>();
 
   useEffect(() => {
-    let tabs: HTMLElement[] = Array.from(ref.current.querySelectorAll('.' + styles['spectrum-Tabs-item']));
-    setSelectedTab(tabs.find((tab) => tab.dataset.key === state.selectedKey));
-  }, [props.children, state.selectedKey]);
+    if (ref.current) {
+      let tabs: HTMLElement[] = Array.from(ref.current.querySelectorAll('.' + styles['spectrum-Tabs-item']));
+      let selectedTab = tabs.find((tab) => tab.dataset.key === state.selectedKey);
+
+      // default to first tab if selectedTab is deleted from children
+      if (selectedTab != null) {
+        setSelectedTab(selectedTab);
+      } else if (tabs[0]) {
+        state.setSelectedKey(tabs[0].dataset.key);
+        setSelectedTab(tabs[0]);
+      }
+    }
+  }, [props.children, state, ref]);
+
+
+  let checkShouldCollapse = useCallback(() => {
+    let getFurthestPoint = (elem) => {
+      if (direction === 'rtl') {
+        return elem.getBoundingClientRect().left;
+      } else {
+        return elem.getBoundingClientRect().right;
+      }
+    };
+
+    let computeShouldCollapse = () => {
```

Comment:
Nah breadcrumbs works in a slightly different way, tabs collapses fully when it can't fit all the tabs horizontally while breadcrumbs collapses only as much as it needs to to fit horizontally